src/todolistconverters.js

```diff
@@ -277,15 +242,55 @@ export function modelViewChangeChecked( onCheckedChange ) {
 		const { mapper, writer: viewWriter } = conversionApi;
 		const isChecked = !!data.item.getAttribute( 'todoListChecked' );
 		const viewItem = mapper.toViewElement( data.item );
-		const itemRange = viewWriter.createRangeIn( viewItem );
-		const oldCheckmarkElement = findInRange( itemRange, item => item.is( 'uiElement' ) ? item : false );
+		// Because of m -> v position mapper we can be sure checkbox is always at the beginning.
+		const oldCheckmarkElement = viewItem.getChild( 0 );
 		const newCheckmarkElement = createCheckmarkElement( data.item, viewWriter, isChecked, onCheckedChange );
 
 		viewWriter.insert( viewWriter.createPositionAfter( oldCheckmarkElement ), newCheckmarkElement );
 		viewWriter.remove( oldCheckmarkElement );
 	};
 }
 
+/**
+ * A model-to-view position at zero offset mapper.
+ *
+ * This helper ensures that position inside todo-list in the view is mapped after the checkbox.
+ *
+ * It only handles the position at the beginning of a list item as other positions are properly mapped be the default mapper.
+ *
+ * @param {module:engine/view/view~View} view
+ * @param {module:engine/conversion/mapper~Mapper} mapper
+ * @return {Function}
+ */
+export function mapModelToViewZeroOffsetPosition( view, mapper ) {
+	return ( evt, data ) => {
+		const modelPosition = data.modelPosition;
+		const parent = modelPosition.parent;
+
+		// Handle only position at the beginning of a todo list item.
+		if ( !parent.is( 'listItem' ) || parent.getAttribute( 'listType' ) != 'todo' || modelPosition.offset !== 0 ) {
+			return;
+		}
+
+		const viewLi = mapper.toViewElement( parent );
+		const label = findLabel( viewLi, view );
+
+		// If there is no label then most probably the default converter was overridden.
+		if ( !label ) {
+			return;
+		}
+
+		// Map the position to the next sibling (if it is not a marker) - most likely it will be a text node...
+		if ( label.nextSibling && !label.nextSibling.is( 'uiElement' ) ) {
```

Comment:
I'm afraid checking only the next element won't be enough. There could be more UIElements at the beginning of the list item. You should walk over them and find a node after the last one.

Reply:
Check the test - it works. We return a position after our UIElement so others are properly (in order of insertion) inserted to the model. For me it is OK.

Reply:
Yep, you are right. It works fine.